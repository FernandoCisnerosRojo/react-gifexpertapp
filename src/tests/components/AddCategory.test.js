// import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory';

describe('AddCategory Tests', () => {
    // const setCategories = () => { };
    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should render "car" from input change', () => {
        const value = 'car'
        const input = wrapper.find('input')

        input.simulate('change', {
            target: {
                value: value
            }
        })

        const p = wrapper.find('p')

        expect(p.text().trim()).toBe(value)
    })

    test('should not call setCategories', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } })

        expect(setCategories).not.toHaveBeenCalled()
    })

    test('should call setCategories once with a function and clean inputValue', () => {
        wrapper.find('input').simulate('change', { target: { value: 'car' } })

        wrapper.find('form').simulate('submit', { preventDefault() { } })

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        // expect(wrapper.find('p').text()).toBe('');
        expect(wrapper.find('input').prop('value')).toBe('');
    })

})
