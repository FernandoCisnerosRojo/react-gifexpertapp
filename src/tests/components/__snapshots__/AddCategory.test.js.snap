// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Tests should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
